src/Kale.hs

```diff
@@ -137,9 +151,9 @@ fileToTask dir file = runMaybeT $
 
 mkTask :: String -> String -> String -> Task
 mkTask fileContent name mod_ = Task
-    { taskModule = mod_
-    , taskArgs   = findArgs fileContent
-    , taskName   = casify name
+    { taskModule = TaskModule mod_
+    , taskArgs   = TaskArgs <$> findArgs fileContent
+    , taskName   = TaskName $ casify name
```

Comment:
I would prefer to see the `mkTask` function take the more strongly typed stuff -- eg `mkTask :: FileContent -> TaskName -> TaskModule -> Task`. Of course, then it's just `Task` :)

We could expose a function `mkTaskName = TaskName . casify` and have that be the only exported function. That way we know that any string in a `TaskName` is already cased appropriately.

Likewise, `mkTaskArgs :: FileContent -> Maybe TaskArgs` could be used as the smart constructor for the task arguments. In fact, we may want to make `TaskArgs` a sum type of `NoArguments | RecordArgs String | PositionalArgs String` :)

Reply:
Just added the changes you requested, let me know what you think.

Reply:
I wasn't sure how to detect if there were record vs positional args.  I took a guess at how to do it, but I don't think it is the best solution.